### ISSUE TITLE
(Auto) fixing invalid icalendar (components)

### ISSUE THREAD

**Issue by geier** [collective/icalendar#223]
As already recently discussed in #222, should we try to fix broken icalendars and/or icalendar components?

The issue in #222 were VTIMEZONE's OFFSETs > 24h which should be minutes (e.g., `TZOFFSETFROM:+5744` =>  `TZOFFSETFROM:+005744`) (I'm inclined to say no, as it's hard to guess what was really meant).

Further issues are VEVENTs without a UID (see https://github.com/pimutils/khal/issues/632) or DTSTART == DTEND.

**Comment by stlaz** [comment]
I don't think we should try to be clever with invalid inputs. After all, we are but a parser, it's our duty to warn the user that their input is not correct, and fail in that case.
We can have some error recovery in case the parsing should not be strict but the best thing we can do is to leave the input untouched, not trying to fix it. The user should fix it themselves in such case. A warning that their input is not correct should still be issued, though.

**Comment by gasull** [comment]
> I don't think we should try to be clever with invalid inputs.

Many [HTML parsers](https://en.wikipedia.org/wiki/Comparison_of_HTML_parsers), like HTML Tidy, attempt to fix invalid HTML, or have the option to do so.  `icalendar` should do the same.
